modules/abstract-eth/src/lib/utils.ts

```diff
@@ -430,23 +442,31 @@ export function decodeNativeTransferData(data: string): NativeTransferData {
 /**
  * Decode the given ABI-encoded flush tokens data and return parsed fields
  *
- * @param data The data to decode
+ * @param txJson The txJson with data to decode
  * @returns parsed transfer data
  */
-export function decodeFlushTokensData(data: string): FlushTokensData {
-  if (!data.startsWith(flushForwarderTokensMethodId)) {
-    throw new BuildTransactionError(`Invalid transfer bytecode: ${data}`);
+export function decodeFlushTokensData(txJson: TxData): FlushTokensData {
+  if (txJson.data.startsWith(flushForwarderTokensMethodId)) {
+    const [forwarderAddress, tokenAddress] = getRawDecoded(
+      flushTokensTypes,
+      getBufferedByteCode(flushForwarderTokensMethodId, txJson.data)
+    );
+    return {
+      forwarderAddress: addHexPrefix(forwarderAddress as string),
+      tokenAddress: addHexPrefix(tokenAddress as string),
+    };
+  } else if (txJson.data.startsWith(flushForwarderTokensMethodIdV4)) {
+    const [tokenAddress] = getRawDecoded(
+      flushTokensTypesv4,
+      getBufferedByteCode(flushForwarderTokensMethodIdV4, txJson.data)
+    );
+    return {
+      forwarderAddress: txJson.to ? txJson.to : '',
```

Comment:
We definitely want to throw a error and stop if we are getting to as empty?

Reply:
Done